static/js/compose.js

```diff
@@ -383,6 +387,17 @@ function validate_stream_message_mentions(stream_name) {
             // user has not seen a warning message yet if undefined
             show_all_everyone_warnings();
 
+            $("#compose-send-button").prop('disabled', false);
+            $("#sending-indicator").hide();
+            return false;
+        }
+    } else if (stream_name === "announce" &&
+        stream_count > compose.announce_warn_threshold) {
+        if (user_acknowledged_all_everyone === undefined ||
+            user_acknowledged_all_everyone === false) {
+            // user has not seen a warning message yet if undefined
+            show_all_everyone_warnings(true);
+
```

Comment:
Together with the change in logic to `show_all_everyone_warnings` proposed above, these if-checks can be simplified: Create a variable like "mention_type" and assign it "all", "everyone", "announce", or "none". Then You can just use one if-statement to make a call `show_warnings(mention_type)`, if `mention_type` is not none.

Reply:
I think the problem here is trying to combine all this logic into one function.  We should just have a separate function for the "announce" codepath.